addons/hr_payroll/report/report_payslip_details.py

```diff
@@ -41,8 +41,9 @@ def get_recursive_parent(rule_categories):
             if not rule_categories:
                 return []
             if rule_categories[0].parent_id:
-                rule_categories.insert(0, rule_categories[0].parent_id)
-                get_recursive_parent(rule_categories)
+                categories = []
+                categories.insert(0, rule_categories[0].parent_id)
+                get_recursive_parent(categories)
```

Comment:
you are doing nothing with `categories` and the result of `get_recursive_parent` is lost